crates/common/mqtt_channel/src/session.rs

```diff
@@ -40,7 +40,11 @@ pub async fn init_session(config: &Config) -> Result<(), MqttError> {
             }
 
             Err(err) => {
-                eprintln!("Connection Error {}", err);
+                if err.to_string().contains("Connection refused") {
+                    eprintln!("Couldn't connect to mqtt broker due to {}", err);
+                } else {
+                    eprintln!("Connection Error {}", err);
+                }
```

Comment:
As you are doing nothing specific to change the error message, why not simply:

```
                    eprintln!("Couldn't connect to mqtt broker due to {}", err);
```

Did you try to translate the error message using the actual message error? This will be less fragile and possibly more helpful. See:

* [ConnectionError](https://docs.rs/rumqttc/latest/rumqttc/enum.ConnectionError.html)
* [ConnectReturnCode](https://docs.rs/rumqttc/latest/rumqttc/mqttbytes/v4/enum.ConnectReturnCode.html)

Reply:
I went with this because I just wanted to address the `mqtt` connection issue when it refused. I feel now it makes sense to just have `  eprintln!("Couldn't connect to mqtt broker due to {}", err);` for all the cases.

Reply:
If there is no `mqtt broker` during `--init` then the error thrown is ` rumqttc::ConnectionError::Io(_)`. So, the `ConnectReturnCode` is not useful.

Reply:
Though since we will be supported authenticated mqtt brokers in the near future, we need to also handle the ConnectReturnCode style errors as well. For example if the broker requires authentication, then it is useful to know if the password is wrong or not, or if it is a protocol version issue etc. These kinds of errors are covered by [ConnectReturnCode](https://docs.rs/rumqttc/latest/rumqttc/mqttbytes/v4/enum.ConnectReturnCode.html#) (as mentioned by @didier-wenzek)

Reply:
Updated the error messages to usethe `ConnectReturnCode`.